.github/workflows/sts-mock-push-to-main.yml

```diff
@@ -0,0 +1,252 @@
+
+    - name: Configure AWS credentials for DEV
+      uses: aws-actions/configure-aws-credentials@55f725fcb586ca16a5ed5b0d75d464defbfa831b #main
+      with:
+        aws-region: eu-west-2
+        role-to-assume: ${{ secrets.STS_MOCK_DEV_GH_ACTIONS_ROLE_ARN }}
+
+    - name: Login to Amazon ECR DEV
+      uses: aws-actions/amazon-ecr-login@a81a5945e74802f35ca53aa274a9e00436e6210e #main
+
+    - name: Push image to DEV
+      run: |
+        docker push $STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+    - name: Install Cosign
+      uses: sigstore/cosign-installer@4959ce089c160fddf62f7b42464195ba1a56d382 #main
+      with:
+        cosign-release: 'v1.9.0'
+
+    - name: Code sign the Docker image
+      id: cosign-image             
+      run: |
+        cosign sign --key awskms:///$DEV_CONTAINER_SIGN_KMS_KEY $STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+  build-and-upload-sam-artifact-to-dev:
+    name: Validate & upload S3 artifact to dev
+    runs-on: ubuntu-latest
+    needs:
+      - build-and-push-test-image-to-dev
+    defaults:
+      run:
+        shell: bash
+        working-directory: sts-mock
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Setup nodeJS v20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: sts-mock/package-lock.json
+
+      - name: Setup SAM CLI
+        uses: aws-actions/setup-sam@2360ef6d90015369947b45b496193ab9976a9b04 #main
+        with:
+          use-installer: true
+          version: 1.123.0
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@55f725fcb586ca16a5ed5b0d75d464defbfa831b #main
+        with:
+          aws-region: eu-west-2
+          role-to-assume: ${{ secrets.STS_MOCK_DEV_GH_ACTIONS_ROLE_ARN }}
+
+      - name: Sam Validate
+        run: |
+          echo "SAM_CLI_TELEMETRY=0" >> $GITHUB_ENV
+          sam validate --lint
+
+      - name: Sam Build
+        run: |
+          sam build --cached
+
+      - name: Upload SAM artifact into the DEV artifact bucket
+        uses: govuk-one-login/devplatform-upload-action@6985ccbaa306e2320a8826252c922af65242b283 #main
+        with:
+          artifact-bucket-name: ${{ secrets.STS_MOCK_DEV_ARTIFACT_BUCKET }}
+          signing-profile-name: ${{ secrets.BACKEND_API_DEV_SIGNING_PROFILE }}
+          template-file: .aws-sam/build/template.yaml
+          working-directory: sts-mock
+
+  build-and-push-test-image-to-build:
+    name: Build and push test image to Build
+    needs: 
+      - sts-mock-tests-and-sonar-scan
+      - build-and-push-test-image-to-dev
+    runs-on: ubuntu-latest
+    env:
+      STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI: ${{ secrets.STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI }}
+      BACKEND_API_BUILD_COSIGN_KEY: ${{secrets.BACKEND_API_BUILD_COSIGN_KEY}}
```

Comment:
It looks like the ENV_VAR is also not needed as it isn't referenced in the following steps. (Unless one of the steps needs it implicitly.) Can we remove it?

Reply:
Done.